slides.md

```diff
@@ -4,7 +4,149 @@
+
+```javascript
+const name = "Daniele";
+
+console.log("Hello " + name);
+console.log('Hello ' + name);
+console.log(`Hello ${name}`);
+```
+
+To create arrays, you can use square brackets `[]`.
+
+```javascript
+const anArray = [1, 2, 3];
+```
+
+To create objects (string indexed dictionaries), you can use curly brackets.
+```javascript
+const obj = {
+  a: 1,
+  b: "Hello",
+  c: {
+    nested: true
+  }
+};
+```
+
+To create a function, you can use both the `function` keyword, or use the arrow `=>` syntax. Arrow functions can directly return a value as an expression, and don't need `()` around a single argument. They need `()` around the return value if they return an object.
+
+TLDR: they're mostly interchangeable, but prefer arrow syntax over function.
+
+Long explanation: The main differences are:
+1. Arrow functions cannot be used as constructors (a function works as a class constructor of itself, kind of, `class` is just syntactical sugar).
+2. `this` in arrow functions is easier to reason about: it's the `this` of the scope in which the function is declared. In a normal function, `this` might change depending on how it's invoked.
+
+```javascript
+function say(name) {
+  console.log(`Hello ${name}`);
+};
+
+const say = (name) => {
+  console.log(`Hello ${name}`);
+};
+
+const double = n => n * 2;
+const makeConfig = () => ({debug: true, log: false}); // returns the object
+```
+
+You can _spread_ objects and arrays, which applies their content to the current expression. The order in which you spread your object **matters**: to decide which ones overrides in case of objects with the same fields, earlier declarations are overridden.
+
+```javascript
+const numbers = [1, 2, 3];
+const add = (a, b, c) => a + b + c;
+const total = add(...numbers); // 6
+const twice = [...numbers, ...numbers]; // [1,2,3,1,2,3]
+
+const defaults = {
+  log: false,
+  debug: true
+}
+
+const options = {
+  debug: false
+}
+
+const overrideOptions = {...options, ...defaults} // {log: false, debug: true}
+const overrideDefaults = {...defaults, ...options} // {log: false, debug: false}
+const overrideWithFields = {...defaults, debug: false} // {log: false, debug: false}
+```
+
+You can **destructure** (access their inner values directly) objects and arrays.
+```javascript
+const [first, second] = [1,2,3]; // first: 1, second: 2
+const [, last] = [1,2,3]; // last: 3
+const add = ([first, second]) => first + second;
+add([1,2]);
+
+const {a, b} = {a: 1, b: 2}; // a: 1, b: 2
+const add = ({first, second}) => first + second;
+add({first: 1, second: 2});
+```
+
+To check for equality, **always** use the triple equal `===`. Objects are **never** equal between each other. `==` casts the two object to the same type before comparing, and assumptions on the behaviour might lead to unexpected results because the implementation might be not intuitive.
```

Comment:
Not clear on what `Objects are **never** equal between each other.` is supposed to mean, here? You mean between different types? Because as-is it reads "equality checks always return false" which is clearly wrong.

Reply:
equality check between 2 different objects (even with the same content) does return always `false`